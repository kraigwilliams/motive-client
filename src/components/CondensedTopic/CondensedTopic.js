import React, { Component } from 'react'
import {Topic, TopicTitle, TopicCount} from './CondensedTopic.style'


class CondensedTopic extends Component {
  render() {

    return (
        <TopicTitle 
          to={`topics/${this.props.id}`}
        >
         <Topic key={this.props.id}>
            {this.props.title}
            <TopicCount>
              {this.props.count}
            </TopicCount>
          </Topic>
        </TopicTitle>
    );
  }
}

export default CondensedTopic