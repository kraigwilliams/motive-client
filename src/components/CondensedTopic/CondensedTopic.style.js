import styled from "styled-components";
import { colors } from "../constants";
import { Link } from "react-router-dom";

export const Topic = styled.div`
  border: 2px solid ${colors.blue};
  padding: 10px 20px 10px 20px;
  color: ${colors.offwhite};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 20px auto;
  border-radius: 10px;
  /* min-width: fit-content; */
  width: 100%;

  :hover {
    box-shadow: 0px 2px 5px 3px rgba(0, 0, 0, 0.5);
  }

  @media (min-width: 1000px) {
    /* min-width: fit-content; */
    width: 80%;
  }

  animation: fadein 2s;
  -moz-animation: fadein 2s; /* Firefox */
  -webkit-animation: fadein 2s; /* Safari and Chrome */
  -o-animation: fadein 2s; /* Opera */

  @keyframes fadein {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
  @-moz-keyframes fadein {
    /* Firefox */
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
  @-webkit-keyframes fadein {
    /* Safari and Chrome */
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
  @-o-keyframes fadein {
    /* Opera */
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
`;

export const TopicTitle = styled(Link)`
  color: ${colors.offwhite};
  font-size: 16px;
  text-decoration: none;
`;

export const TopicCount = styled.p`
  color: ${colors.offwhite};
  font-size: 16px;
  margin-left: 10px;
`;
