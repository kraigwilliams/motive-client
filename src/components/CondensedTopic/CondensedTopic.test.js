import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import CondensedTopic from "./CondensedTopic";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe(`<CondensedTopic />`, () => {
  //Smoke Testing
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Router>
        <CondensedTopic />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  //Snapshot Testing
  it("renders CondensedTopic by default", () => {
    const wrapper = shallow(
      <Router>
        <CondensedTopic />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
