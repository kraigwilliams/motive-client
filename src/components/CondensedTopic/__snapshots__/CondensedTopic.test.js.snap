// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CondensedTopic /> renders CondensedTopic by default 1`] = `
<styled.div>
  <Styled(Link)
    to="topics/undefined"
  />
  <styled.p />
</styled.div>
`;
